tools/report-converter/codechecker_report_converter/sparse/output_parser.py

```diff
@@ -0,0 +1,99 @@
+# -------------------------------------------------------------------------
+#
+#  Part of the CodeChecker project, under the Apache License v2.0 with
+#  LLVM Exceptions. See LICENSE for license information.
+#  SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# -------------------------------------------------------------------------
+
+import logging
+import os
+import re
+
+from ..output_parser import BaseParser, Message, Event
+LOG = logging.getLogger('ReportConverter')
+
+
+class SparseParser(BaseParser):
+    """
+    Parser for Sparse Output
+    """
+
+    def __init__(self, analyzer_result):
+        super(SparseParser, self).__init__()
+
+        self.analyzer_result = analyzer_result
+
+        self.message_line_re = re.compile(
+            # File path followed by a ':'.
+            r'^(?P<path>[\S ]+?):'
+            # Line number followed by a ':'.
+            r'(?P<line>\d+?):'
+            # Column number followed by a ':'.
+            r'(?P<column>\d+?):'
+            # Message.
+            r'(?P<message>[\S \t]+)\s*')
+
+        self.multi_line_message_re = re.compile(
+            # File path followed by a ':'.
+            r'^(?P<path>[\S ]+?):'
+            # Message.
+            r'(?P<message>[\S \t]+)\s*'
+        )
+
+        self.note_line_re = re.compile(
+            # File path followed by a ':'.
+            r'^(?P<path>\.[\S ]+?):'
+            # Line number followed by a ':'.
+            r'(?P<line>\d+?):'
+            # Column number followed by a ':'.
+            r'(?P<column>\d+?):'
+            # Message.
+            r'(?P<message>[\S \t]+)\s*'
+        )
+
+    def parse_message(self, it, line):
```

Comment:
I tried to convert what the sample.out with the report-converter tool and then store the results to a running server. Now there is only one report with a lot of bug steps:
![image](https://user-images.githubusercontent.com/6695818/108853508-1844f680-75e7-11eb-8688-474a4a276c26.png)
I don't think this is what you want.
In my previous comment (https://github.com/Ericsson/codechecker/pull/3160#discussion_r578315134) my recommendation was the following:
- Every line which looks like this will be a new report (Message): `./files/sample.h:3:40: warning: incorrect type in argument 2 (different address spaces)` (the `warning` keyword identifies that we have a new report).
- Every line which looks like this will be an event / bug step item of the previous Message object: `./files/sample.h:3:40:    expected struct task_struct *p1`. (So it doesn't contain the warning keyword like in the previous case).
- Every line can be skipped which looks like this `files/sample.c: note: in included file:`.

So in case of the following output:
```
files/sample.c:4:1: warning: symbol 'machine_id' was not declared. Should it be static?
files/sample.h:3:40: warning: incorrect type in argument 2 (different address spaces)
files/sample.c: note: in included file:
files/sample.h:3:40: warning: incorrect type in argument 1 (different address spaces)
files/sample.h:3:40:    expected struct task_struct *p1
files/sample.h:3:40:    got struct spinlock [noderef] __rcu *
```
I want to see 3 reports:
1. files/sample.c:4:1: warning: symbol 'machine_id' was not declared. Should it be static?
2. files/sample.h:3:40: warning: incorrect type in argument 2 (different address spaces)
3. files/sample.h:3:40: warning: incorrect type in argument 1 (different address spaces)
files/sample.h:3:40:    expected struct task_struct *p1
files/sample.h:3:40:    got struct spinlock [noderef] __rcu *

In the third case the report will have multiple bug steps.

Reply:
> I tried to convert what the sample.out with the report-converter tool and then store the results to a running server. Now there is only one report with a lot of bug steps:
> ![image](https://user-images.githubusercontent.com/6695818/108853508-1844f680-75e7-11eb-8688-474a4a276c26.png)
> I don't think this is what you want.

Well Actually I thought that is what we wanted, a single report for all the files and hence changed the code to include it in the single report. 😢 


> In my previous comment ([#3160 (comment)](https://github.com/Ericsson/codechecker/pull/3160#discussion_r578315134)) my recommendation was the following:
> 
> * Every line which looks like this will be a new report (Message): `./files/sample.h:3:40: warning: incorrect type in argument 2 (different address spaces)` (the `warning` keyword identifies that we have a new report).

This logic won't always stand true as some lines also start with `error`. Do you recommend modifying the regex as well now?


> * Every line which looks like this will be an event / bug step item of the previous Message object: `./files/sample.h:3:40:    expected struct task_struct *p1`. (So it doesn't contain the warning keyword like in the previous case).
> * Every line can be skipped which looks like this `files/sample.c: note: in included file:`.
> 
> So in case of the following output:
> 
> ```
> files/sample.c:4:1: warning: symbol 'machine_id' was not declared. Should it be static?
> files/sample.h:3:40: warning: incorrect type in argument 2 (different address spaces)
> files/sample.c: note: in included file:
> files/sample.h:3:40: warning: incorrect type in argument 1 (different address spaces)
> files/sample.h:3:40:    expected struct task_struct *p1
> files/sample.h:3:40:    got struct spinlock [noderef] __rcu *
> ```
> 
> I want to see 3 reports:
> 
> 1. files/sample.c:4:1: warning: symbol 'machine_id' was not declared. Should it be static?
> 2. files/sample.h:3:40: warning: incorrect type in argument 2 (different address spaces)
> 3. files/sample.h:3:40: warning: incorrect type in argument 1 (different address spaces)
>    files/sample.h:3:40:    expected struct task_struct *p1
>    files/sample.h:3:40:    got struct spinlock [noderef] __rcu *
> 
> In the third case the report will have multiple bug steps.

Alright so in the tests, we will also have two `sample.c.expected.plist` and `sample.h.expected.plist`, this is after the assumption that second line in the `sample.out` file is removed. Correct me if I am wrong